sdk/storage/azure-storage-blob/V8_V12.md

```diff
@@ -0,0 +1,301 @@
+```java
+BlobContainerClient client = new BlobContainerClientBuilder()
+    .connectionString("${connection-string}")
+    .containerName("mycontainer")
+    .buildClient();
+```
+
+### Shared key credential
+Shared key credentials can be used directly to authenticate your client. You need to fetch the `${accountName}` and `${accountKey}` from Azure Portal.
+Learn More from [README](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob/README.md)
+
+In V8, you used the credential to build the `CloudStorageAccount`.
+```java
+// Initialize the credentials
+StorageCredentialsAccountAndKey credential = new StorageCredentialsAccountAndKey("${accountName}", "${accountKey}");
+// CloudStorageAccount take the credential.
+CloudStorageAccount storageAccount = new CloudStorageAccount(credential);
+```
+
+Since the builder in V12 manages pipeline generation, you can just hand the `StorageSharedKeyCredential` to the builder.
+```java
+// Initialize the credentials
+StorageSharedKeyCredential credential = new StorageSharedKeyCredential("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://" + "<account-name>" + ".blob.core.windows.net")
+    .credential(credential)
+    .buildClient();
+```
+
+### SASToken
+A URL with a SAS token looks like the following: `https://${accountName}.blob.core.windows.net/?${sasToken}`.
+Refer the [documentation](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob/README.md) for more info.
+
+See section [Generate SAS token](#generate-sas-token) for how to generate the token programmatically.
+
+In V8, you had to sasToken string, which can paste from portal or generate from session above .
+```java
+// Here is how the blob object take in the sas token generated in above section [Generate SAS token](#generate-sas-token).
+StorageCredentialsSharedAccessSignature credential = new StorageCredentialsSharedAccessSignature("sasToken");
+CloudStorageAccount storageAccount = new CloudStorageAccount(credential);
+```
+
+In V12, SAS tokens can be managed just like any other credential.
+```java
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net")
+    .sasToken("${sasToken}")
+    .buildClient();
+```
+
+However, if you already have the full URL with SAS token attached, you can use that too:
+```java
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://${accountName}.blob.core.windows.net/?${sasToken}")
+    .buildClient();
+```
+
+## Minimum Overload APIs and Maxmum Overload APIs:
+In V8, we only provide one maximum overload API for each operation. 
```

Comment:
This statement isn't accurate. v8 had tons and tons of overloads

Reply:
Changed the description.